### ISSUE TITLE
Url parsing mangles domain name

### ISSUE THREAD

**Issue by max-te** [tiged/tiged#70]
I'm trying to use tiged on a repository hosted on a private gitlab instance with a domain name like `gitlab.developers.example.de`. Unfortunately tiged breaks the domain name:

```
$ npx tiged https://gitlab.developers.example.de/foo/bar
! Command failed: git clone --depth 1 https://gitlabvelopers.example.de.de/foo/bar .
Cloning into '.'...
fatal: unable to access 'https://gitlabvelopers.example.de.de/foo/bar/': Failed to connect to gitlabvelopers.example.de.de port 443 after 40 ms: Couldn't connect to server
```

**Comment by kands-code** [comment]
I've tried `https://gitlab.gnome.org/GNOME/gnome-2048`, and it works fine.

![截图 2023-04-06 10-42-17](https://user-images.githubusercontent.com/59006329/230259266-789e23e5-47aa-46a4-a439-3667d4adf387.png)

Are you sure you have access to the target repository? Please give an example.

**Comment by max-te** [comment]
I am not free to share the actual url i was working with when experiencing, but I can assure you that I can `git clone` from that url just fine. I crafted the example url above to reproduce the error. You can see the mangling in the `Command failed` output. Somehow it's triggered by the `.de`: once in `.developers` and once as the tld.

Similar mangling does not happen in these examples:
```
npx tiged https://gitlab.developers.example.com/foo/bar
! Command failed: git clone --depth 1 https://gitlab.developers.example.com/foo/bar .
```

```
npx tiged https://gitlab.dovelopers.example.de/foo/bar
! Command failed: git clone --depth 1 https://gitlab.dovelopers.example.de/foo/bar .
```

**Comment by nake89** [comment]
I have replicated this problem. There is indeed and issue with the parse() function. I will look into this and try to fix it.

**Comment by nake89** [comment]
The fix is quite simple actually. What mangles it is the way degit handles TLDs (in a very bad way)

```js
const siteName = tld ? site.replace(tld, '') : site;
```
See the problem? If there is a domain that has .tld as a subdomain (even part of it) e.g .developers. It will mangle it.

I replaced it with:
```js
const siteName = tld ? site.replace(new RegExp(tld + '$'), '') : site;
```

Automated tests worked. I wanna do some manual tests before I publish a new version.

**Comment by nake89** [comment]
Fixed in newest version 2.12.5
